### ISSUE TITLE
`no-undefined-types` - false positive on `@link` to arrow function argument

### ISSUE THREAD

**Issue by DetachHead** [gajus/eslint-plugin-jsdoc#1704]
thanks for the quick fix for #1701!

i found another similar issue though, the rule does not seem to work with arrow functions.

## Expected behavior
no error

## Actual behavior
```
The type 'foo' is undefined. (jsdoc/no-undefined-types)
```

## ESLint Config

```js
/** @type {import('eslint').Linter.Config} */
const config = {
  extends: [
    'plugin:@typescript-eslint/recommended',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
    project: './tsconfig.json',
  },
  rules: {
    'import/no-undefined-types': 'error',
  },
};

module.exports = config;
```

## ESLint sample

```ts
// no error
/** {@link foo} */
function a(foo: number) {}

// error: The type 'foo' is undefined. (false positive)
/** {@link foo} */
const b = (foo: number) => {};
```

this is a false positive because go to definition on `foo` works in both cases

## Environment

- Node version: 22.17.0
- ESLint version 8.57.1
- `eslint-plugin-jsdoc` version: 63.0.3
